reverie/backend_server/persona/prompt_template/run_gpt_prompt.py

```diff
@@ -359,63 +359,122 @@ def create_prompt_input(persona, task, duration, test_input=None):
-                _cr += [" ".join([j.strip() for j in i.split(" ")][3:])]
+                actions_without_the_enumeration_at_the_start += [" ".join([j.strip() for j in i.split(" ")][3:])]
             else:
-                _cr += [i]
-        for count, i in enumerate(_cr):
-            k = [j.strip() for j in i.split("(duration in minutes:")]
-            task = k[0]
-            if task[-1] == ".":
-                task = task[:-1]
-            duration = int(k[1].split(",")[0].strip())
-            cr += [[task, duration]]
+                actions_without_the_enumeration_at_the_start += [i]
+
+        for count, i in enumerate(actions_without_the_enumeration_at_the_start):
+            # extract duration in minutes
+            activity_duration_temporary = [j.strip() for j in i.split("(duration in minutes:")]
+            activity_temp = activity_duration_temporary[0]
+
+            # remove full stop
+            if activity_temp[-1] == ".":
+                activity_temp = activity_temp[:-1]
+
+            # get the duration
+            duration = int(activity_duration_temporary[1].split(",")[0].strip())
+
+            activities_and_durations += [[activity_temp, duration]]
 
+        # get how long the task was supposed to last
         total_expected_min = int(prompt.split("(total duration in minutes")[-1]
                                  .split("):")[0].strip())
 
         # TODO -- now, you need to make sure that this is the same as the sum of
         #         the current action sequence.
-        curr_min_slot = [["dummy", -1], ]  # (task_name, task_index)
-        for count, i in enumerate(cr):
-            i_task = i[0]
+        accumulates_all_activites = [["dummy", -1], ]  # (task_name, task_index)
+
+        for activity_number, i in enumerate(activities_and_durations):
+            i_activity = i[0]
             i_duration = i[1]
 
+            # just normalize to a multiple of 5 e.g., 23%5 = 3 so 23 - 3 = 20
             i_duration -= (i_duration % 5)
+
+            # guess this filters out bad activities and adding one activity
+            # accumulates_all_activites is going to contain N activites where N is the
+            # number of minutes (if gpt was good gpt)
             if i_duration > 0:
                 for j in range(i_duration):
-                    curr_min_slot += [(i_task, count)]
-        curr_min_slot = curr_min_slot[1:]
+                    accumulates_all_activites += [(i_activity, activity_number)]
+
+        # remove... dummy?
+        accumulates_all_activites = accumulates_all_activites[1:]
 
-        if len(curr_min_slot) > total_expected_min:
-            last_task = curr_min_slot[60]
+        # if we gpt was bad, we might have more activities than the required minutes
+        if len(accumulates_all_activites) > total_expected_min:
+            # we get the last task...
+            # is this right? maybe the variable name is just wrong
+            last_task = accumulates_all_activites[60]
```

Comment:
60 may not be the last one, if gpt was good it should be accumulates_all_activites [total_expected_min] but have no guarantee